src/server/svr_jobfunc.c

```diff
@@ -5783,20 +5782,20 @@ find_ms_full_host_and_port(job *pjob, int *port)
 
 #define	LOG_ERR_BUF(err_msg, err_msg_sz, msg, errno, header_msg) \
 if ((err_msg != NULL) && (err_msg_sz > 0)) { \
-	snprintf(err_msg, err_msg_sz-1, msg); \
+	snprintf(err_msg, err_msg_sz, msg); \
 	log_err(errno, header_msg, err_msg); \
 }
 
 #define	LOG_EVENT_BUF_ARG1(err_msg, err_msg_sz, msg, arg, id) \
 if ((err_msg != NULL) && (err_msg_sz > 0)) { \
-	snprintf(err_msg, err_msg_sz-1, msg, arg); \
+	snprintf(err_msg, err_msg_sz, msg, arg); \
 	log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_DEBUG, \
 					id, err_msg); \
 }
 
 #define	LOG_EVENT_BUF_ARG3(err_msg, err_msg_sz, msg, arg1, arg2, arg3, id) \
 if ((err_msg != NULL) && (err_msg_sz > 0)) { \
-	snprintf(err_msg, err_msg_sz-1, msg, arg1, arg2, arg3); \
+	snprintf(err_msg, err_msg_sz, msg, arg1, arg2, arg3); \
 	log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_DEBUG, \
 					id, err_msg); \
 }
```

Comment:
Oh wait, is that the guts LOG_EVENT_BUF_ARG3 macro?  Then yes, you absolutely need them.  I thought it was just random code with \\'s at the end.